executor/include/execution/trait/can_prefer.hpp

```diff
@@ -5,27 +5,37 @@
 #pragma once
 
 #include <execution/trait/can_require.hpp>
+#include <execution/trait/common_traits.hpp>
 #include <type_traits>
 
 namespace execution {
 
 template <typename Executor, typename Property,
           typename std::enable_if_t<
               Property::template is_applicable_property_v<Executor> &&
-                  Property::is_preferable,
+                  Property::is_preferable && can_require_v<Executor, Property>,
               int> = 0>
-Executor prefer(Executor ex, const Property t) {
-  if (execution::can_require_v<Executor, Property>) return ex.require(t);
-  return ex;
+constexpr auto prefer(Executor &&ex, const Property &p) noexcept {
+  return ex.require(p);
+}
+
+template <typename Executor, typename Property,
+          typename std::enable_if_t<
+              Property::template is_applicable_property_v<Executor> &&
+                  Property::is_preferable && !can_require_v<Executor, Property>,
+              int> = 0>
+constexpr auto prefer(Executor &&ex, const Property &p) noexcept {
+  return std::forward<Executor>(ex);
```

Comment:
In order for `std::forward` the return type of this functions needs to be `auto&&`

Reply:
`decltype(auto)` would also work right?

Reply:
If you want to be sneaky, putting parens `return (...);` also works with `auto` as return type :rofl: